extensions/dacpac/src/wizard/dataTierApplicationWizard.ts

```diff
@@ -288,47 +301,152 @@ export class DataTierApplicationWizard {
-		const service = await this.getService(msSqlProvider);
-		const ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
+		const importStartTime = new Date().getTime();
+		let service: mssql.IDacFxService;
+		let ownerUri: string;
+		let result: mssql.DacFxResult;
+		let additionalProps: { [key: string]: string } = {};
+		let additionalMeasurements: { [key: string]: number } = {};
+		try {
+			service = await this.getService(msSqlProvider);
+			ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
+			result = await service.importBacpac(this.model.filePath, this.model.database, ownerUri, azdata.TaskExecutionMode.execute);
+		} catch (e) {
+			additionalProps.exceptionOccured = 'true';
+		}
 
-		return await service.importBacpac(this.model.filePath, this.model.database, ownerUri, azdata.TaskExecutionMode.execute);
+		// If result is null which means exception occured, will be adding additional props to the Telemetry
+		if (!result) {
+			additionalProps = { ...additionalProps, ...this.getDacServiceArgsAsProps(service, this.model.database, this.model.filePath, ownerUri) };
+		}
+		additionalProps.importStatus = result?.success.toString();
+		additionalMeasurements.importedBacpacFileSizeBytes = await utils.tryGetFileSize(this.model.filePath);
+		additionalMeasurements.totalDurationMs = (new Date().getTime() - importStartTime);
+		// Import Bacpac: 'Import button' clicked in Import summary page, Reporting the event selection to the telemetry
+		this.DacServiceTelemetry(TelemetryViews.ImportBacpac, 'ImportBacpacOperation', additionalProps, additionalMeasurements);
+		return result;
 	}
 
 	public async generateDeployScript(): Promise<mssql.DacFxResult> {
-		const service = await this.getService(msSqlProvider);
-		const ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
-		this.wizard.message = {
-			text: loc.generatingScriptMessage,
-			level: azdata.window.MessageLevel.Information,
-			description: ''
-		};
-
-		let result = await service.generateDeployScript(this.model.filePath, this.model.database, ownerUri, azdata.TaskExecutionMode.script);
+		const genScriptStartTime = new Date().getTime();
+		let service: mssql.IDacFxService;
+		let ownerUri: string;
+		let result: mssql.DacFxResult;
+		let additionalProps: { [key: string]: string } = {};
+		let additionalMeasurements: { [key: string]: number } = {};
+		try {
+			this.wizard.message = {
+				text: loc.generatingScriptMessage,
+				level: azdata.window.MessageLevel.Information,
+				description: ''
+			};
+
+			service = await this.getService(msSqlProvider);
+			ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
+			result = await service.generateDeployScript(this.model.filePath, this.model.database, ownerUri, azdata.TaskExecutionMode.script);
+		} catch (e) {
+			additionalProps.exceptionOccured = 'true';
```

Comment:
`exceptionOccurred`

Reply:
Oh! you are too quick @Charles-Gagnon, :) I was wondering how did i miss this comment😁